.github/workflows/full_stack_tests.yml

```diff
@@ -221,21 +221,21 @@ jobs:
           | tee desktop_acceptance_test_${{ steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME }}.log
 
       - name: Upload desktop test logs
-        if: ${{ steps.desktop-test-attempt-1.outcome == 'failure' }}
+        if: ${{ failure() && steps.desktop-test-attempt-1.outcome == 'failure' }}
```

Comment:
Can you confirm what `steps.desktop-test-attempt-1.outcome == 'failure'` outputs? Is this actually the correct way to check for failure?

More generally, what is the difference between failure() and the other clause? I think we need to properly understand this (and simplify if needed) if we want to keep this code maintainable.

Reply:
@seanlip Here is screenshot of before when none of three (failure, success, always) was passed and github action automatically added success() to if condition.

<img width="672" height="165" alt="image" src="https://github.com/user-attachments/assets/09b35ae4-51ca-4f26-a772-932245569cf6" />